### DISCUSSION TITLE
Auto merge isn't working?

### DISCUSSION THREAD

**Discussion by PythonCoderAS** [community/community#31782] [Pull Requests]
I followed the directions in https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request, but instead of enabling auto-merge, GitHub Actions instantly merges the PR even without checking if all checks resolved.

**Comment by airtower-luna** [comment]
Do you have a branch protection rule that marks the checks you want as required for merging? Auto-merge waits only for those.

**Reply by PythonCoderAS** [reply]
Oh. I thought it would work regardless. Do I need to enable auto-merge in the settings? It's currently disabled. And I don't really have any specific checks that I want to succeed, I just want all possible checks to pass.

**Reply by airtower-luna** [reply]
> Do I need to enable auto-merge in the settings? It's currently disabled.

That too, yes.

> And I don't really have any specific checks that I want to succeed, I just want all possible checks to pass.

Then just set all of them as required in the branch protection rule. Basically what auto-merge does is "merge as soon as applicable branch-protection rules allow". :slightly_smiling_face:

**Reply by PythonCoderAS** [reply]
If I add new checks in the future, do I need to re-do branch protection as well?

**Reply by airtower-luna** [reply]
Not fully redo, but update, yes. You can add and remove required checks for an existing branch protection rule.
